homeassistant/components/honeywell/climate.py

```diff
@@ -517,7 +521,8 @@ async def _login() -> None:
         except UnauthorizedError:
             await _login()
             return
-
+        except APIRateLimited:
```

Comment:
Why would you just return instead of following the retry logic for other errors? Seems like the intent of that logic is to say after a certain amount of failures you want to mark this as unavailable. If you're getting continual API rate limit errors, would that not apply?

Reply:
The API has a login rate limit https://github.com/mkmer/AIOSomecomfort/blob/aa49c5e45f4e2c7ae8cd96ff314d26ca84a7d523/aiosomecomfort/__init__.py#L82-L85
When the login is limited, update was receiving this uncaught exception:
- Refresh triggered
- Update fails due to Unauthorized
- Login is triggered (and at rate limit)
- Login returns `APIRateLimited`
- refresh gets hit with uncaught `APIRateLimited`.

I've made a change to the API to return `APIRateLimited` when refresh is triggered and login is limited rather than fail a refresh and try to login.  

I suppose we could mark it unavailable, but it will always become unavailable as the rate limit is longer than the `refresh rate*_retry` with a bit of delay.